packages/common/src/index.ts

```diff
@@ -401,18 +402,29 @@ export default class Common extends EventEmitter {
     if (typeof chain === 'number' || typeof chain === 'string' || BN.isBN(chain)) {
       // Filter out genesis states if passed in to customChains
       let plainCustomChains: IChain[]
-      if (
-        this._customChains &&
-        this._customChains.length > 0 &&
-        Array.isArray(this._customChains[0])
-      ) {
-        plainCustomChains = (this._customChains as [IChain, GenesisState][]).map((e) => e[0])
+      if (this._customChains?.length && Array.isArray(this._customChains[0])) {
+        plainCustomChains = (this._customChains as [IChain, GenesisState | GethGenesisState][]).map(
+          ([chain, state]) => {
+            if ('hash' && 'stateRoot' in state) {
```

Comment:
i believe this needs to be
```suggestion
            if ('hash' in state && 'stateRoot' in state) {
```

Reply:
a trick i use when the list starts getting long (from parse.ts):

```typescript
if (['config', 'difficulty', 'gasLimit', 'alloc'].some((field) => !(field in json))) {
  throw new Error('Invalid format, expected geth genesis fields missing')
}
```